tests/test_dsl.py

```diff
@@ -343,3 +343,220 @@ def assert_cohorts_equivalent(dsl_cohort, qm_cohort):
+    series1 = date_series()
+    series2 = date_series()
+
+    expected = Comparator(
+        operator="__le__",
+        lhs=series1.value,
+        rhs=Comparator(lhs=series2.value, operator="__ne__"),
+    )
+
+    output = series1 <= series2
+    assert repr(output) == repr(BoolSeries(value=expected))
+
+
+def test_dateseries_ne(date_series):
+    series1 = date_series()
+    series2 = date_series()
+
+    expected = Comparator(
+        operator="__ne__",
+        lhs=series1.value,
+        rhs=Comparator(lhs=series2.value, operator="__ne__"),
+    )
+
+    output = series1 != series2
+    assert repr(output) == repr(BoolSeries(value=expected))
+
+
+def test_intseries_gt(int_series):
+    series1 = int_series()
+    series2 = int_series()
+
+    expected = Comparator(
+        operator="__gt__",
+        lhs=series1.value,
+        rhs=Comparator(lhs=series2.value, operator="__ne__"),
+    )
+
+    output = series1 > series2
+    assert repr(output) == repr(BoolSeries(value=expected))
+
+
+def test_intseries_ge(int_series):
+    series1 = int_series()
+    series2 = int_series()
+
+    expected = Comparator(
+        operator="__ge__",
+        lhs=series1.value,
+        rhs=Comparator(lhs=series2.value, operator="__ne__"),
+    )
+
+    output = series1 >= series2
+    assert repr(output) == repr(BoolSeries(value=expected))
+
+
+def test_intseries_lt(int_series):
+    series1 = int_series()
+    series2 = int_series()
+
+    expected = Comparator(
+        operator="__lt__",
+        lhs=series1.value,
+        rhs=Comparator(lhs=series2.value, operator="__ne__"),
+    )
+
+    output = series1 < series2
+    assert repr(output) == repr(BoolSeries(value=expected))
+
+
+def test_intseries_le(int_series):
+    series1 = int_series()
+    series2 = int_series()
+
+    expected = Comparator(
+        operator="__le__",
+        lhs=series1.value,
+        rhs=Comparator(lhs=series2.value, operator="__ne__"),
+    )
+
+    output = series1 <= series2
+    assert repr(output) == repr(BoolSeries(value=expected))
+
+
+def test_intseries_ne(int_series):
+    series1 = int_series()
+    series2 = int_series()
+
+    expected = Comparator(
+        operator="__ne__",
+        lhs=series1.value,
+        rhs=Comparator(lhs=series2.value, operator="__ne__"),
+    )
+
+    output = series1 != series2
+    assert repr(output) == repr(BoolSeries(value=expected))
+
+
+def test_patientseries_eq(patient_series):
+    series1 = patient_series()
+    series2 = patient_series()
+
+    expected = Comparator(
+        operator="__eq__",
+        lhs=series1.value,
+        rhs=Comparator(lhs=series2.value, operator="__ne__"),
+    )
+
+    output = series1 == series2
+    assert repr(output) == repr(BoolSeries(value=expected))
+
+
+def test_patientseries_invert():
+    series = ~PatientSeries(value=Comparator())
+    assert isinstance(series.value, Comparator)
+
+    series = ~PatientSeries(value=Value())
+    assert isinstance(series.value, Comparator)
```

Comment:
Any reason this test doesn't use the same structure as those above?

Reply:
nope, just missed that test, good catch!